### ISSUE TITLE
Proposed solution makes link rot problem worse; points to non-navigational parts

### ISSUE THREAD

**Issue by ylebre** [WICG/ScrollToTextFragment#3]
The proposed scroll to text fragment URLs will be even more prone to link rot than normal links already are. In the case of normal (fragment) links, the owner of the webpage is in control of which fragments are available. When changing the page, they could to the diligent work and make sure all the expected links work as intended and either provide the original information, or a suggestion where the information has moved.

With the proposed solution, this is not the case - the proposed fragment links point to an arbitrary part of the webpage where the owner of the website has no idea these links even exist so there is no way the owner can keep the links working.

In the example:
www.example.com#targetText=the%20lazy%20dog,brown%20fox

Would match "The lazy dog jumped over the quick brown fox". However, if the owner of the site decided to add another snippit of text ("the lazy dog has a brown coat, and it looks just like a brown fox") that also matches the targetText, the link would be invalid and would actually point to another part of the document not intended by the link.

In the example pointing to PDF pages, audio tracks etc:
The difference with links in media fragments and PDFs, is that in those cases, the underlying content is a lot less likely to change. Links to a specific page number, track or row are all pointing to a part of the document or media that are already numbered. These numbers are already there with the intent of allowing (human) navigation to them, so it would stand to reason that linking to them would also be a good idea. Text passages in a web page are not intended as part of the navigational structure of the page, which makes linking to them unexpected behaviour.

tl;dr:
- linking should point to things that are part of the navigational structure. This makes it OK to point to chapters, pages, sections, rows etc, but not OK to link to fragments of text.
- the owner of a page should have control on what links "work" on that page. This ensures that they have the ability to make changes to the page without breaking any links to the page.

**Comment by kevinmarks** [comment]
This is a mistaken objection. The text of a page is far more resilient than the structure as templates are changed often. More importantly, the goal of linking is most often to refer to a particular piece of text, not some page structure. By embedding the text in the link this makes it more resilient, not less as the intent of the linker is clearer.
See [how to present links as quotes](http://www.kevinmarks.com/mentionquote.html) as an example.

**Comment by bokand** [comment]
In addition to @kevinmarks' point above, the updated proposal now supports adding context snippets which can prevent the given example ("the lazy dog has a brown coat, and it looks just like a brown fox").

**Comment by ylebre** [comment]
I think adding context could help make more specific links, but that would also make the links even more brittle as they now depend on other pieces as well. The proposal should work fine on immutable content, but content behind a URL can be modified over time. The original author does not know that their content is being linked this way, so they can't redirect the faulty link to the new location of the original content.

**Comment by bokand** [comment]
Sure, but presumably if the content moves, the page can redirect. Perhaps we can make the fragment survive a 302 in which case everything world work. Otherwise the user simply gets to the targetted resource without the targetText which is the current status quo.

Content behind a link can already change - at least with this proposal, we could surface the fact that the referenced text no longer exists on the page. In the worst case, we get today's behavior of simply navigating to the stale content. I fail to see how this makes anything worse...

**Comment by tilgovi** [comment]
We should not forget that the use cases are potentially different. If the intention is to link to a publisher-provided identifier then the publisher can make those identifiers known, e.g. when GitHub sets the fragment to match the line you click in a file view.

Linking to text provides something the publisher has not specifically identified. That puts power in the hands of the content consumer. While it may be fragile, it's not something a consumer can do _at all_ now.

This is not a choice between two alternatives.

Let's also keep in mind that just because "rot" might have negative connotations, link rot is not "bad", per se. The web relies upon the fragility of reference. A link might break, but the page containing it does not. When you convey a reference, you are conveying it with all its fragility and that's okay.

**Comment by bokand** [comment]
I tend to agree with the rationales above, mainly:

1) Including the text in the link means the text can be surfaced to the user, even if it no longer appears in the page.
2) This provides some additional context - in the worst case, we fallback to existing behavior so it's strictly not-worse.

The discussion was useful but I'm doing a clean-up pass over issues so I'm going to close this as there's nothing actionable here.
